astacus/node/download.py

```diff
@@ -134,7 +134,8 @@ def _cb(*, map_in: Sequence[ipc.SnapshotFile], map_out: Sequence[ipc.SnapshotFil
                 while chown_candidate != self.dst and chown_candidate.stat().st_uid != dst_owner_uid:
                     chowned_paths.add(str(chown_candidate.relative_to(self.dst)))
                     chown_candidate = chown_candidate.parent
-            if chowned_paths:
+
+            for chowned_path in chowned_paths:
                 astacus_user = getpass.getuser()
                 dst_owner_gid = self.dst.stat().st_gid
                 # We're very specific to allow a sufficiently restrictive sudoers configuration
```

Comment:
If the list is very long, we probably don't want to spawn thousand of subprocesses,
one for every single file, a chunked approach might be better:

```python
                # We're very specific to allow a sufficiently restrictive sudoers configuration
                chunk_size = 1000
                sorted_paths = sorted(chowned_paths)
                for chunk_start in range(0, len(sorted_paths), chunk_size):
                    cmd: Sequence[str | Path] = [
                        "/usr/bin/sudo",
                        "/usr/bin/chown",
                        f"--from={astacus_user}:{dst_owner_gid}",
                        f"{dst_owner_uid}",
                        "--",
                        *sorted_paths[chunk_start: chunk_start + chunk_size],
                    ]
                    subprocess.run(cmd, shell=False, check=True, cwd=self.dst)
```